README
README
------

The Galette projected:
http://galette.tuxfamily.org

If you have questions or need support, please use mailing list:
https://mail.gna.org/listinfo/galette-discussion


Le projet Galette :
http://galette.tuxfamily.org

Si vous avez des questions, ou si vous avez besoin d'aide, contactez
la liste de diffusion :
https://mail.gna.org/listinfo/galette-discussion


ChangeLog
* 2016-10-18 - 1.6.2
- Galette 0.8.3 changes

* 2015-04-26 - 1.6
- Fix fields configuration reset
- Drop encoding conversion

* 2014-09-14 - 1.5
- Galette 0.8 compatibility

* 2013-08-10 - 1.4
- Galette 0.7.5 compatibility
- Add PDF models reinitialization
- Add PHP modules check

* 2013-05-07 - 1.3
- Add fields configuration reset
- Fix typos
- Menu entry was not selected

* 2013-01-06 - 1.2
- Galette 0.7.3 compatibility
- Update © date
- Add fields reinitialization button
- Add a function to fill empty logins and passwords

* 2012-10-06 - 1.1
- Galette 0.7.1 compatibility
- minor changges

* 2011-12-15 - 1.0
- First stable release


_smarties.php
<?php

/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */

/**
 * Configuration file for AdminTools plugin
 *
 * PHP version 5
 *
 * Copyright © 2011-2014 The Galette Team
 *
 * This file is part of Galette (http://galette.tuxfamily.org).
 *
 * Galette is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Galette is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Galette. If not, see <http://www.gnu.org/licenses/>.
 *
 * @category  Plugins
 * @package   GaletteAdminTools
 *
 * @author    Johan Cwiklinski <johan@x-tnd.be>
 * @copyright 2011-2014 The Galette Team
 * @license   http://www.gnu.org/licenses/gpl-3.0.html GPL License 3.0 or (at your option) any later version
 * @version   SVN: $Id$
 * @link      http://galette.tuxfamily.org
 * @since     Available since 0.7dev - 2011-11-21
 */

$_tpl_assignments = array(
    'admintools_tpl_dir' => '__plugin_templates_dir__',
    'admintools_dir'     => '__plugin_dir__'
);

?>


admintools.php
<?php

/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */

/**
 * Administration tools
 *
 * This page can be loaded directly, or via ajax.
 * Via ajax, we do not have a full html page, but only
 * that will be displayed using javascript on another page
 *
 * PHP version 5
 *
 * Copyright © 2011-2014 The Galette Team
 *
 * This file is part of Galette (http://galette.tuxfamily.org).
 *
 * Galette is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Galette is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Galette. If not, see <http://www.gnu.org/licenses/>.
 *
 * @category  Plugins
 * @package   GalettePaypal
 * @author    Johan Cwiklinski <johan@x-tnd.be>
 * @copyright 2011-2014 The Galette Team
 * @license   http://www.gnu.org/licenses/gpl-3.0.html GPL License 3.0 or (at your option) any later version
 * @version   SVN: $Id: owners.php 556 2009-03-13 06:48:49Z trashy $
 * @link      http://galette.tuxfamily.org
 * @since     Available since 0.7dev - 2011-11-21
 */

use Galette\Entity\Adherent;
use Galette\Entity\FieldsConfig;
use Galette\Entity\Texts;
use Galette\Repository\Members;
use Galette\Repository\PdfModels;

define('GALETTE_BASE_PATH', '../../');

require_once GALETTE_BASE_PATH . 'includes/galette.inc.php';

//Constants and classes from plugin
require_once '_config.inc.php';

//restricted to superadmin only
if ( !$login->isSuperAdmin() ) {
    header('location: ' . GALETTE_BASE_PATH . 'index.php');
    die();
}

$error_detected = array();
$success_detected = array();

if ( isset($_POST['inittexts']) ) {
    //proceed mails texts reinitialization
    $texts = new Texts($texts_fields, $preferences);
    $res = $texts->installInit(false);
    if ( $res === true ) {
        $success_detected[] = _T("Texts has been successfully reinitialized.");
    } else {
        $error_detected[] = _T("An error occured reinitializing texts :(");
    }
}

if ( isset($_POST['initfields']) ) {
    //proceed fields configuration reinitialization
    include_once GALETTE_BASE_PATH . 'includes/fields_defs/members_fields.php';
    include_once GALETTE_BASE_PATH . 'includes/fields_defs/members_fields_cats.php';
    $fc = new FieldsConfig($zdb, Adherent::TABLE, $members_fields, $members_fields_cats, true);
    $res = $fc->installInit();
    if ( $res === true ) {
        $success_detected[] = _T("Fields configuration has been successfully reinitialized.");
    } else {
        $error_detected[] = _T("An error occured reinitializing fields configuration :(");
    }
}

if ( isset($_POST['initpdfmodels']) ) {
    //proceed mails texts reinitialization
    $models = new PdfModels($zdb, $preferences);
    $res = $models->installInit($pdfmodels_fields, false);
    if ( $res === true ) {
        $success_detected[] = _T("PDF models has been successfully reinitialized.");
    } else {
        $error_detected[] = _T("An error occured reinitializing PDF models :(");
    }
}

if ( isset($_POST['emptylogins']) ) {
    //proceed empty logins and passwords
    //those ones cannot be null
    $members = new Members();
    $res = $members->emptylogins();
    if ( $res === true ) {
        $success_detected[] = str_replace(
            '%i',
            $members->getCount(),
            _T("Logins and passwords has been successfully filled (%i processed).")
        );
    } else {
        $error_detected[] = _T("An error occured filling empty logins and passwords :(");
    }
}

if ( isset($_POST['checkmodules']) ) {
    $cm = new Galette\Core\CheckModules();
    $modules_ok = $cm->isValid();
    if ( !$modules_ok ) {
        $error_detected[] = _T("Some PHP modules are missing. Please install them or contact your support.<br/>More informations on required modules may be found in the documentation.");
    }
    $modules_res = $cm->toHtml();
    $tpl->assign('modules_res', $modules_res);
    $tpl->assign('require_dialog', true);
}

//Set the path to the current plugin's templates,
//but backup main Galette's template path before
$orig_template_path = $tpl->template_dir;
$tpl->template_dir = 'templates/' . $preferences->pref_theme;
$tpl->compile_id = ADMINTOOLS_SMARTY_PREFIX;
//set util paths
$plugin_dir = basename(dirname($_SERVER['SCRIPT_NAME']));
$tpl->assign(
    'galette_url',
    'http://' . $_SERVER['HTTP_HOST'] .
    preg_replace(
        "/\/plugins\/" . $plugin_dir . '/',
        "/",
        dirname($_SERVER['SCRIPT_NAME'])
    )
);
$tpl->assign(
    'plugin_url',
    'http://' . $_SERVER['HTTP_HOST'] . dirname($_SERVER['SCRIPT_NAME']) . '/'
);
$tpl->assign('page_title', _T("Administration tools"));
if ( count($error_detected) > 0 ) {
    $tpl->assign('error_detected', $error_detected);
}
if ( count($success_detected) > 0 ) {
    $tpl->assign('success_detected', $success_detected);
}
$content = $tpl->fetch('admintools.tpl', ADMINTOOLS_SMARTY_PREFIX);
$tpl->assign('content', $content);
//Set path back to main Galette's template
$tpl->template_dir = $orig_template_path;
$tpl->display('page.tpl', ADMINTOOLS_SMARTY_PREFIX);


_config.inc.php
<?php

/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */

/**
 * Configuration file for AdminTools plugin
 *
 * PHP version 5
 *
 * Copyright © 2011-2014 The Galette Team
 *
 * This file is part of Galette (http://galette.tuxfamily.org).
 *
 * Galette is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Galette is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Galette. If not, see <http://www.gnu.org/licenses/>.
 *
 * @category  Plugins
 * @package   GaletteAdminTools
 *
 * @author    Johan Cwiklinski <johan@x-tnd.be>
 * @copyright 2011-2014 The Galette Team
 * @license   http://www.gnu.org/licenses/gpl-3.0.html GPL License 3.0 or (at your option) any later version
 * @version   SVN: $Id$
 * @link      http://galette.tuxfamily.org
 * @since     Available since 0.7dev - 2011-11-21
 */

define('ADMINTOOLS_SMARTY_PREFIX', 'plugins_admintools');
?>


lang/lang_french.php
<?php
// This file was automatically generated on Sat May 18 21:38:05 2013 (GMT)
// Don't modify it by hand, rather use the target lang from the Makefile.


// ../templates/default/admintools.tpl:6 ../templates/default/admintools.tpl:29 
$lang['Make sure you\'ve done a backup of the database before using any of the following tools!'] = 'Assurez-vous de faire une sauvegarde de votre base de données avant d\'utiliser l\'un des outils suivants&nbsp;!';

// ../templates/default/admintools.tpl:10 
$lang['Reinitialize emails contents'] = 'Réinitialiser les contenus des emails';

// ../templates/default/admintools.tpl:39 
$lang['Convert encoding'] = 'Conversion de l\'encodage';

// ../admintools.php:99 
$lang['Logins and passwords has been successfully filled (%i processed).'] = 'Les login et mot de passe ont été correctement remplis (%i traités).';

// ../templates/default/admintools.tpl:36 
$lang['That conversion stuff will read your entire database, and write back each non UTF-8 values after conversion; this may take a while.'] = 'Cette conversion va lire l\'intégralité de votre base de données, et ré-écrire chaque valeur non UTF-8 après conversion. Cela peut prendre un certain temps.';

// ../admintools.php:86 
$lang['An error occured reinitializing fields configuration :('] = 'Une erreur est survenue lors de la réinitialisation de la configuration des champs :(';

// ../templates/default/menu.tpl:4 
$lang['Admin tools'] = 'Outils admin';

// ../templates/default/admintools.tpl:11 
$lang['Reinitialize fields configuration'] = 'Réinitialiser la configuration des champs';

// ../admintools.php:74 
$lang['An error occured reinitializing texts :('] = 'Une erreur est survenue lors de la réinitialisation des textes :(';

// ../admintools.php:64 
$lang['Database should have been successfully converted to UTF-8!'] = 'La base de données devrait avoir été convertie en UTF-8&nbsp;!';

// ../templates/default/admintools.tpl:33 
$lang['That function is available for MySQL installations only!'] = 'Cette fonction n\'est disponible que pour les installations sous MySQL&nbsp;!';

// ../templates/default/admintools.tpl:34 
$lang['Converts the whole database data to UTF-8, do not change anything on the tables, just proceed data conversion.'] = 'Converti l\'ensemble des données de la base  en UTF-8, ne change rien sur les tables, effectue juste la conversion.';

// ../admintools.php:110 
$lang['Some PHP modules are missing. Please install them or contact your support.<br/>More informations on required modules may be found in the documentation.'] = 'Certains modules PHP sont manquants. Veuillez les installer ou contacter votre support. <br/>Plus d\'informations sur les modules requis sont disponibles dans la documentation.';

// ../templates/default/admintools.tpl:21 
$lang['Check PHP modules'] = 'Vérification des modules PHP';

// ../templates/default/admintools.tpl:27 
$lang['Database tools'] = 'Outils de base de données';

// ../admintools.php:72 
$lang['Texts has been successfully reinitialized.'] = 'Les textes ont été réinitialisés avec succès.';

// ../templates/default/admintools.tpl:4 
$lang['Content tools'] = 'Outils de contenus';

// ../admintools.php:137 
$lang['Administration tools'] = 'Outils d\'administration';

// ../templates/default/admintools.tpl:47 
$lang['Modules report'] = 'Rapport sur les modules';

// ../templates/default/admintools.tpl:16 
$lang['Generate values for empty logins and passwords'] = 'Génère des valeurs pour les login et mot de passe vides';

// ../templates/default/admintools.tpl:35 
$lang['You should use that function if you can see strange characters in your datas; after a manual import, or if you\'ve not used Galette\'s update scripts.'] = 'Vous devriez utiliser cette fonctions si d\'étranges caractères aparaissent dans vos données ; après un import manuel, ou si vous n\'avez pas utilisé les scripts de mise à jour de Galette.';

// ../templates/default/menu.tpl:4 
$lang['Various administrative tools'] = 'Divers outils administratifs';

// ../admintools.php:84 
$lang['Fields configuration has been successfully reinitialized.'] = 'La configuration des champs a été réinitialisée avec succès.';

// ../admintools.php:102 
$lang['An error occured filling empty logins and passwords :('] = 'Une erreur est survenue lors de la création de valeurs pour les identifiants vides :(';

// ../templates/default/menu.tpl:2 
$lang['Admin Tools'] = 'Outils Admin';

?>

lang/lang_english.php
<?php
// This file was automatically generated on Sat May 18 21:38:05 2013 (GMT)
// Don't modify it by hand, rather use the target lang from the Makefile.


// ../templates/default/admintools.tpl:6 ../templates/default/admintools.tpl:29 
$lang['Make sure you\'ve done a backup of the database before using any of the following tools!'] = 'Make sure you\'ve done a backup of the database before using any of the following tools!';

// ../templates/default/admintools.tpl:10 
$lang['Reinitialize emails contents'] = 'Reinitialize emails contents';

// ../templates/default/admintools.tpl:39 
$lang['Convert encoding'] = 'Convert encoding';

// ../admintools.php:99 
$lang['Logins and passwords has been successfully filled (%i processed).'] = 'Logins and passwords has been successfully filled (%i processed).';

// ../templates/default/admintools.tpl:36 
$lang['That conversion stuff will read your entire database, and write back each non UTF-8 values after conversion; this may take a while.'] = 'That conversion stuff will read your entire database, and write back each non UTF-8 values after conversion; this may take a while.';

// ../admintools.php:86 
$lang['An error occured reinitializing fields configuration :('] = 'An error occured reinitializing fields configuration :(';

// ../templates/default/menu.tpl:4 
$lang['Admin tools'] = 'Admin tools';

// ../templates/default/admintools.tpl:11 
$lang['Reinitialize fields configuration'] = 'Reinitialize fields configuration';

// ../admintools.php:74 
$lang['An error occured reinitializing texts :('] = 'An error occured reinitializing texts :(';

// ../admintools.php:64 
$lang['Database should have been successfully converted to UTF-8!'] = 'Database should have been successfully converted to UTF-8!';

// ../templates/default/admintools.tpl:33 
$lang['That function is available for MySQL installations only!'] = 'That function is available for MySQL installations only!';

// ../templates/default/admintools.tpl:34 
$lang['Converts the whole database data to UTF-8, do not change anything on the tables, just proceed data conversion.'] = 'Converts the whole database data to UTF-8, do not change anything on the tables, just proceed data conversion.';

// ../admintools.php:110 
$lang['Some PHP modules are missing. Please install them or contact your support.<br/>More informations on required modules may be found in the documentation.'] = 'Some PHP modules are missing. Please install them or contact your support.<br/>More informations on required modules may be found in the documentation.';

// ../templates/default/admintools.tpl:21 
$lang['Check PHP modules'] = 'Check PHP modules';

// ../templates/default/admintools.tpl:27 
$lang['Database tools'] = 'Database tools';

// ../admintools.php:72 
$lang['Texts has been successfully reinitialized.'] = 'Texts has been successfully reinitialized.';

// ../templates/default/admintools.tpl:4 
$lang['Content tools'] = 'Content tools';

// ../admintools.php:137 
$lang['Administration tools'] = 'Administration tools';

// ../templates/default/admintools.tpl:47 
$lang['Modules report'] = 'Modules report';

// ../templates/default/admintools.tpl:16 
$lang['Generate values for empty logins and passwords'] = 'Generate values for empty logins and passwords';

// ../templates/default/admintools.tpl:35 
$lang['You should use that function if you can see strange characters in your datas; after a manual import, or if you\'ve not used Galette\'s update scripts.'] = 'You should use that function if you can see strange characters in your datas; after a manual import, or if you\'ve not used Galette\'s update scripts.';

// ../templates/default/menu.tpl:4 
$lang['Various administrative tools'] = 'Various administrative tools';

// ../admintools.php:84 
$lang['Fields configuration has been successfully reinitialized.'] = 'Fields configuration has been successfully reinitialized.';

// ../admintools.php:102 
$lang['An error occured filling empty logins and passwords :('] = 'An error occured filling empty logins and passwords :(';

// ../templates/default/menu.tpl:2 
$lang['Admin Tools'] = 'Admin Tools';

?>

_define.php
<?php

/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */

/**
 * Configuration file for AdminTools plugin
 *
 * PHP version 5
 *
 * Copyright © 2011-2014 The Galette Team
 *
 * This file is part of Galette (http://galette.tuxfamily.org).
 *
 * Galette is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Galette is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Galette. If not, see <http://www.gnu.org/licenses/>.
 *
 * @category  Plugins
 * @package   GaletteAdminTools
 *
 * @author    Johan Cwiklinski <johan@x-tnd.be>
 * @copyright 2011-2014 The Galette Team
 * @license   http://www.gnu.org/licenses/gpl-3.0.html GPL License 3.0 or (at your option) any later version
 * @version   SVN: $Id$
 * @link      http://galette.tuxfamily.org
 * @since     Available since 0.7dev - 2011-11-21
 */

$this->register(
    'Galette Admin Tools',      //Name
    'Administration tools',     //Short description
    'Johan Cwiklinski',         //Author
    '1.6',                      //Version
    '0.8',                      //Galette compatible version
    '2014-09-14',               //Release date
    null                        //Permissions needed - not yet implemented
);

?>
